server/src/core/server/app/handlers/api/story/count.ts

```diff
@@ -198,42 +208,53 @@ export const countsV2Handler =
 
       const { storyIDs }: CountsV2Body = validate(CountsV2BodySchema, req.body);
 
-      const storyCounts = await Promise.all(
-        storyIDs.map(async (storyID) => {
-          // check that cache is available
-
-          // try to look in Redis cache here first by key
-          const key = `${tenant.id}:${storyID}:count`;
-          const redisCount = await redis.get(key);
-
-          if (redisCount) {
-            logger.debug("found story count for counts v2 in redis cache", {
-              storyID,
-              redisCount,
-            });
-            return { storyID, redisCount };
-          } else {
-            const count = await retrieveStoryCommentCounts(
-              mongo,
-              tenant.id,
-              storyID
-            );
-
-            // add count to Redis cache here then return
-            await redis.set(key, count);
-            logger.debug("set story count for counts v2 in redis cache", {
-              storyID,
-              count,
-            });
-
-            return {
-              storyID,
-              count,
-            };
-          }
-        })
+      const redisCounts = await redis.mget(
+        ...storyIDs.map((id) => computeCountKey(tenant.id, id))
```

Comment:
redis really likes it when you ask for everything all at once, it's scheduler is great at grabbing n-many keys and returning it in a fast serialized payload!